buddypress/groups/single/group.php

```diff
@@ -47,7 +53,728 @@
             $verified = false;
         default: 
             $verified = false;
-    }
+	}
+
+	$search_user = isset($_GET['u']) && strlen(trim($_GET['u'])) > 0 ? trim($_GET['u']) : false;
+	if (
+		isset($search_user) && 
+		(strpos($search_user, '"') !== false || 
+		strpos($search_user, "'") !== false || 
+		strpos($search_user, '\\') !== false)
+	) {
+		$search_user = str_replace('\\', '', $search_user);
+		$search_user = preg_replace('/^\"|\"$|^\'|\'$/', "", $search_user);
+	}
+	
+    $first_name = false;
+    $last_name = false;
+
+    // We aren't searching a username rather a full name
+    if($search_user && strpos($search_user, ' ') !== false) {
+        $name = explode(' ', $search_user);
+        if(is_array($name) && sizeof($name) === 2) {
+            $first_name = $name[0];
+            $last_name = $name[1];
+        }
+	}
+
+    $country_code = isset($_GET['location']) && strlen($_GET['location']) > 0 ? strtoupper(trim($_GET['location'])) : false;
+    $get_tag = isset($_GET['tag']) && strlen(trim($_GET['tag'])) > 0 ? strtolower(trim($_GET['tag'])) : false;
+    $language_code = isset($_GET['language']) && strlen($_GET['language']) > 0 ? strtolower(trim($_GET['language'])) : false;
+	
+	$filtered_members = Array();
+    $used_country_list = Array();
+    $used_languages = Array();
+
+    // Time to filter stuff
+    foreach($members['members'] AS $index => $member) {
```

Comment:
I think we should put a sanity check here on the members index.  Make sure it's set and it's an array.  I don't envision a time when it wouldn't be I think we should still make the check.